src/agent/agent.js

```diff
@@ -198,7 +198,9 @@ class Agent {
         );
 
         // register rpc n2n
-        this.n2n_agent = this.rpc.register_n2n_agent(this.client.node.n2n_signal);
+        this.n2n_agent = this.rpc.register_n2n_agent((...args) =>
```

Comment:
(style) I would keep those one liners if not too long, because these are just a small technicality.

Reply:
I think this makes it much less readable. but if you insist I will change

Reply:
done